pkg/instancetype/preference/requirements/cpu.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+package requirements
+
+import (
+	"fmt"
+
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	"kubevirt.io/api/instancetype/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/instancetype/apply"
+	preferenceApply "kubevirt.io/kubevirt/pkg/instancetype/preference/apply"
+)
+
+const (
+	InsufficientInstanceTypeCPUResourcesErrorFmt = "insufficient CPU resources of %d vCPU provided by instance type, preference requires " +
+		"%d vCPU"
+	InsufficientVMCPUResourcesErrorFmt = "insufficient CPU resources of %d vCPU provided by VirtualMachine, preference " +
+		"requires %d vCPU provided as %s"
+	NoVMCPUResourcesDefinedErrorFmt = "no CPU resources provided by VirtualMachine, preference requires %d vCPU"
+)
+
+func (h *Handler) checkCPU() (apply.Conflicts, error) {
+	if h.instancetypeSpec != nil {
+		if h.instancetypeSpec.CPU.Guest < h.preferenceSpec.Requirements.CPU.Guest {
```

Comment:
Do we want to check this requirement also against VMs where resources were provided without an instancetype?